core/new-gui/src/app/dashboard/component/user-avatar/user-avatar.component.ts

```diff
@@ -0,0 +1,62 @@
+import { GooglePeopleApiResponse } from "./../../type/google-api-response";
+import { Component, OnInit, Input } from "@angular/core";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+import { HttpClient } from "@angular/common/http";
+import { environment } from "../../../../environments/environment";
+
+@UntilDestroy()
+@Component({
+  selector: "texera-user-avatar",
+  templateUrl: "./user-avatar.component.html",
+  styleUrls: ["./user-avatar.component.scss"],
+})
+
+/**
+ * UserAvatarComponent is used to show the avatar of a user
+ * The avatar of a Google user will be its Google profile picture
+ * The avatar of a normal user will be a default one with the initial
+ * 
+ * Use Google People API to retrieve google user's profile picture
+ * Check https://developers.google.com/people/api/rest/v1/people/get for more details of the api usage
+ * 
+ * @author Zhen Guan
+ */
+export class UserAvatarComponent implements OnInit {
+  public googleUserAvatarSrc: string = "";
+  private publicKey = environment.google.publicKey;
+
+  constructor(private http: HttpClient) {}
+
+  @Input() googleId?: string;
+  @Input() userName?: string;
+
+  ngOnInit(): void {
+    if (!this.googleId && !this.userName) {
+      throw new Error("google Id or user name should be provided");
+    } else if (this.googleId) {
+      this.userName = "";
+      // get the avatar of the google user
+      const googlePeopleAPIUrl = `https://people.googleapis.com/v1/people/${this.googleId}?personFields=names%2Cphotos&key=${this.publicKey}`;
```

Comment:
does this google avatar contain style already? for example bg color

Reply:
yes, it is the same avatar that the google user has set in his/her google account